### PR TITLE
refactor hilt stuff and rip dependencies out of view

### PR DESCRIPTION
**Pull request by plusmobileapps**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-    id 'kotlin-android-extensions'
     id 'kotlin-kapt'
     id 'dagger.hilt.android.plugin'
 }
 
 android {
-    compileSdkVersion 30
-    buildToolsVersion "30.0.2"
+    compileSdkVersion 31
+//    buildToolsVersion "31.0.0"
```
Comment (plusmobileapps) [inline app/build.gradle:10]: not sure this is needed and just update the whole project to use the new gradle plugin, java 11, and all the latest dependency versions

File: app/src/main/java/co/grove/storefinder/DependencyInjectionModule.kt [hunk 2]
```diff
@@ -1,37 +1,22 @@
...
 import dagger.Provides
 import dagger.hilt.InstallIn
-import dagger.hilt.android.components.ApplicationComponent
 import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
 import javax.inject.Singleton
 
 @Module
-@InstallIn(ApplicationComponent::class)
+@InstallIn(SingletonComponent::class)
```
Comment (plusmobileapps) [inline app/src/main/java/co/grove/storefinder/DependencyInjectionModule.kt:14]: component was renamed to singleton meaning it is the application scope essentially

File: app/src/main/java/co/grove/storefinder/model/StoreRepo.kt [hunk 3]
```diff
@@ -5,16 +5,13 @@ import kotlinx.coroutines.launch
...
-class StoreRepo(private val inputStream: InputStream) {
-    private lateinit var storeList: List<Store>
-
-    private suspend fun loadStoreData() = suspendCoroutine<Unit> {
-        val reader = BufferedReader(InputStreamReader(inputStream))
-        val csvFile = CSVFile(reader)
-        storeList = csvFile.parseStoreList()
-    }
+@Singleton
+class StoreRepo @Inject constructor(private val storeCache: StoreCache) {
```
Comment (plusmobileapps) [inline app/src/main/java/co/grove/storefinder/model/StoreRepo.kt:13]: You can annotate within the class its a singleton and also inject dependencies already declared in the graph with inject.

File: app/src/main/java/co/grove/storefinder/model/StoreRepo.kt [hunk 4]
```diff
@@ -5,16 +5,13 @@ import kotlinx.coroutines.launch
...
+    private var storeList: List<Store> = emptyList()
```
Comment (plusmobileapps) [inline app/src/main/java/co/grove/storefinder/model/StoreRepo.kt:14]: we can probably talk about making this reactive with a kotlin flow in coroutines

File: app/src/main/java/co/grove/storefinder/viewmodel/MainViewModel.kt [hunk 5]
```diff
@@ -20,44 +24,41 @@ enum class Units(var displayString: String) {
     }
 }
 
-class MainViewModel : ViewModel() {
-    var units = MutableLiveData<Units>()
-    var addressField = MutableLiveData<String>()
+sealed class MainState {
```
Comment (plusmobileapps) [inline app/src/main/java/co/grove/storefinder/viewmodel/MainViewModel.kt:27]: ditched the interface and represented the state as a sealed class.

File: app/src/main/java/co/grove/storefinder/viewmodel/MainViewModel.kt [hunk 6]
```diff
@@ -20,44 +24,41 @@ enum class Units(var displayString: String) {
...
+    object EmptyAddress : MainState()
+    data class Error(val message: String) : MainState()
+    data class StoreFound(val store: Store, val distance: Double, val units: Units) : MainState()
+}
 
-    lateinit var networkManager: NetworkManager
-    lateinit var activityInterface: MainActivityInterface
-    lateinit var closestStoreFinder: ClosestStoreFinder
+@HiltViewModel
```
Comment (plusmobileapps) [inline app/src/main/java/co/grove/storefinder/viewmodel/MainViewModel.kt:33]: hilt view model and inject annotaion allow you to use the `by viewModels()` function in the view and have hilt automattically inject the dependencies it knows are in the graph.

**Review by MobileOak — APPROVED** [review]

**Pull request merged**